src/Psl/DateTime/DateTimeConvenienceMethodsTrait.php

```diff
@@ -403,28 +403,29 @@ public function plusMonths(int $months): static
             return $this->minusMonths(-$months);
         }
 
-        $current_year = $this->getYear();
-        $current_month = $this->getMonthEnum();
-        $days_to_add = 0;
-        for ($i = 0; $i < $months; $i++) {
-            $total_months = $current_month->value + $i;
-            $target_year = $current_year + Math\div($total_months - 1, MONTHS_PER_YEAR);
-            $target_month = $total_months % MONTHS_PER_YEAR;
-            if ($target_month === 0) {
-                $target_month = 1;
-            }
-
-            $days_to_add += Month::from($target_month)->getDaysForYear($target_year);
+        $plus_years = intdiv($months, 12);
+        $months_left = $months - ($plus_years * 12);
+        $target_month = $this->getMonth() + $months_left;
+
+        if ($target_month > 12) {
+            $plus_years++;
+            $target_month = $target_month - 12;
         }
 
-        return $this->plus(Duration::days($days_to_add));
+        return $this->withDate(
+            $target_year = $this->getYear() + $plus_years,
+            $target_month = Month::from($target_month)->value,
+            min([
+                $this->getDay(),
+                Month::from($target_month)->getDaysForYear($target_year)
```

Comment:
we could create the enum in the line before to avoid having two calls to `Month::from`